### PR TITLE
Refactor all tests to minimize global state

### PR DESCRIPTION
**Pull request by jamilbk**
Tests were intermittently failing for two reasons:

* We were modifying the Application env with `restore_env/3` (a custom function) between tests. This is not thread-safe.
* The Conf cache was also being modified between tests. This is a global memory store and is also not thread-safe.

This PR aims to mock both of these using Mox in order to provide for a manageable way to test the repurcussions of both Application env and Conf cache changes.

### CODE REVIEW

File: apps/fz_http/test/fz_http/telemetry_test.exs [hunk 1]
```diff
@@ -40,119 +40,115 @@ defmodule FzHttp.TelemetryTest do
...
-
     test "count openid providers" do
       ping_data = Telemetry.ping_data()
 
       assert ping_data[:openid_providers] == 7
     end
 
-    @tag config: {:disable_vpn_on_oidc_error, true}
     test "disable vpn on oidc error enabled" do
+      stub_app_env(:disable_vpn_on_oidc_error, true)
```
Comment (akrousset) [inline apps/fz_http/test/fz_http/telemetry_test.exs]: Should this be `stub_conf` like in the `false` test case below?

Comment (jamilbk) [reply apps/fz_http/test/fz_http/telemetry_test.exs]: Good catch!

**Comment by jamilbk** [comment]
@conectado @akrousset -- updated. Though I'm consistently getting weird errors now when running locally, probably due to the concurrency. Going to open another PR to get the device IP sequence fixed.

```
08:57:00.637 [error] Postgrex.Protocol (#PID<0.907.0>) disconnected: ** (DBConnection.ConnectionError) client #PID<0.1789.0> timed out because it queued and checked out the connection for longer than 15000ms

Client #PID<0.1789.0> is still using a connection from owner at location:

    :prim_inet.recv0/3
    (postgrex 0.16.5) lib/postgrex/protocol.ex:3171: Postgrex.Protocol.msg_recv/4
    (postgrex 0.16.5) lib/postgrex/protocol.ex:2201: Postgrex.Protocol.recv_bind/3
    (postgrex 0.16.5) lib/postgrex/protocol.ex:2126: Postgrex.Protocol.bind_execute/4
    (ecto_sql 3.9.1) lib/ecto/adapters/sql/sandbox.ex:375: Ecto.Adapters.SQL.Sandbox.Connection.proxy/3
    (db_connection 2.4.3) lib/db_connection/holder.ex:354: DBConnection.Holder.holder_apply/4
    (db_connection 2.4.3) lib/db_connection.ex:1413: DBConnection.run_execute/5
    (db_connection 2.4.3) lib/db_connection.ex:1508: DBConnection.run/6
    (db_connection 2.4.3) lib/db_connection.ex:644: DBConnection.parsed_prepare_execute/5
    (db_connection 2.4.3) lib/db_connection.ex:636: DBConnection.prepare_execute/4
    (postgrex 0.16.5) lib/postgrex.ex:340: Postgrex.query/4
    (ecto_sql 3.9.1) lib/ecto/adapters/sql.ex:955: Ecto.Adapters.SQL.struct/10
    (ecto 3.9.2) lib/ecto/repo/schema.ex:756: Ecto.Repo.Schema.apply/4
    (ecto 3.9.2) lib/ecto/repo/schema.ex:369: anonymous fn/15 in Ecto.Repo.Schema.do_insert/4
    (fz_http 0.0.0+git.0.deadbeef) lib/fz_http/devices.ex:68: FzHttp.Devices.create_device/1
    (fz_http 0.0.0+git.0.deadbeef) test/support/fixtures/devices_fixtures.ex:26: FzHttp.DevicesFixtures.device/1
    (fz_http 0.0.0+git.0.deadbeef) test/support/test_helpers.ex:146: anonymous fn/1 in FzHttp.TestHelpers.create_rules/1
    (elixir 1.14.2) lib/enum.ex:1662: anonymous fn/3 in Enum.map/2
    (elixir 1.14.2) lib/enum.ex:4299: Enum.reduce_range/5
    (elixir 1.14.2) lib/enum.ex:2472: Enum.map/2

The connection itself was checked out by #PID<0.1789.0> at location:

    (postgrex 0.16.5) lib/postgrex.ex:340: Postgrex.query/4
    (ecto_sql 3.9.1) lib/ecto/adapters/sql.ex:955: Ecto.Adapters.SQL.struct/10
    (ecto 3.9.2) lib/ecto/repo/schema.ex:756: Ecto.Repo.Schema.apply/4
    (ecto 3.9.2) lib/ecto/repo/schema.ex:369: anonymous fn/15 in Ecto.Repo.Schema.do_insert/4
    (fz_http 0.0.0+git.0.deadbeef) lib/fz_http/devices.ex:68: FzHttp.Devices.create_device/1
    (fz_http 0.0.0+git.0.deadbeef) test/support/fixtures/devices_fixtures.ex:26: FzHttp.DevicesFixtures.device/1
    (fz_http 0.0.0+git.0.deadbeef) test/support/test_helpers.ex:146: anonymous fn/1 in FzHttp.TestHelpers.create_rules/1
    (elixir 1.14.2) lib/enum.ex:1662: anonymous fn/3 in Enum.map/2
    (elixir 1.14.2) lib/enum.ex:4299: Enum.reduce_range/5
    (elixir 1.14.2) lib/enum.ex:2472: Enum.map/2
    (fz_http 0.0.0+git.0.deadbeef) test/support/test_helpers.ex:141: FzHttp.TestHelpers.create_rules/1
    FzHttp.UsersTest.__ex_unit_describe_18/1
    (ex_unit 1.14.2) lib/ex_unit/runner.ex:505: ExUnit.Runner.exec_test_setup/2
```

File: apps/fz_http/lib/actual/app.ex [hunk 2]
```diff
@@ -0,0 +1,9 @@
+defmodule Actual.Application do
+  @moduledoc """
+  The opposite of Wrapped.App: Calls the actual Application and Cache modules.
```
Comment (jamilbk) [inline apps/fz_http/lib/actual/app.ex]:
```suggestion
  The opposite of Wrapped.Application: Calls the actual Application modules.
```

File: apps/fz_http/lib/fz_http/telemetry.ex [hunk 3]
```diff
@@ -109,7 +110,7 @@ defmodule FzHttp.Telemetry do
 
   defp common_fields do
     [
-      distinct_id: conf(:telemetry_id),
+      distinct_id: Application.fetch_env!(:fz_http, :telemetry_id),
```
Comment (akrousset) [inline apps/fz_http/lib/fz_http/telemetry.ex]: Should this be `app()`?

**Review by conectado — APPROVED** [review]
LGTM

File: apps/fz_http/lib/fz_http/telemetry.ex (hunk 3 above)
Comment (jamilbk) [inline apps/fz_http/lib/fz_http/telemetry.ex]:
```suggestion
      distinct_id: app().fetch_env!(:fz_http, :telemetry_id),
```

**Pull request merged**